Bestpractices.md

```diff
@@ -8,15 +8,45 @@ Whether you are building an App for your own company or for all of Zendesk custo
+* DRY. Be critical of your own app, if you see that you've repeatedly done the same thing in your code then look to simplify by making helper functions that can be used in more than one place.
+
+* Cleanup. Use life-cycle events `app.deactivated` and `app.willDestroy` to tidy up your app's state.  The intent for these events, and a general best practice, is to cleanup anything that your app might have set up that is no longer needed and might impact on performance of Lotus/other apps.  A prime candidate for cleanup would be intervals you've created via `setInterval` or `setTimeout` (worth noting that app Developers should always store a reference to an ID returned from either of the above methods so as to be able to call `clearInterval` or `clearTimeout`.
+
+* Use promises to handle asynchronous instructions. See [JavaScript Promises: an Introduction](https://developers.google.com/web/fundamentals/getting-started/primers/promises) on Google's developer portal for a great introduction to JavaScript promises.
+
+* Cache the result of promises and API requests when the data doesn't change often.
+
+* Use [secure settings](https://developer.zendesk.com/apps/docs/apps-v2/using_sdk#using-secure-settings) for passwords and API tokens.
+
+* Check out our [changelog](https://developer.zendesk.com/apps/docs/apps-v2/changelog) frequently to keep up to date with the latest framework updates.
+
+* When [creating a ticket in telephony apps](https://support.zendesk.com/entries/24539263#topic_o32_xv1_sk), be sure that you're setting the via_id to 44 (voicemail), 45 (inbound call), or 46 (outbound call) depending on the type of call. This ensures that Zendesk admins and agents are able to properly report on these tickets within Zendesk.
+
+## v2 Guidelines
+
+### Best Practices
+
+* Prefer [bulk calls](https://developer.zendesk.com/apps/docs/apps-v2/using_sdk#bulk-calls) over single calls whenever possible.
+
+* Specify `autoLoad: false` for locations where you don't need to display a user interface and use the `background` location with the Instances API to interact with those locations. See [Instances API Sample App](https://github.com/zendesk/demo_apps/tree/master/v2/support/instances_sample_app) for an example of this technique.
+
+* Use the [App Scaffold](https://github.com/zendesk/app_scaffold/tree/from-scratch)'s from-scratch branch when starting development of any non-trivial app. The App Scaffold includes many features to help you maintain and scale your app.
+
+* Use [signed urls](https://developer.zendesk.com/apps/docs/apps-v2/using_sdk#authenticating-requests-with-signed-urls) when developing server-side apps.
+
+## v1 Guidelines
+
+### Must avoid
+
+:no_entry: In this section you'll find practices you must avoid if you want your app to pass the approval process :no_entry:
 
 * Do not try to create your own [Handlebars helper](http://handlebarsjs.com/), it is not possible at this stage.
```

Comment:
Remove at this stage? Maybe say because it could cause collisions with other apps / the product?